qa/tasks/cephfs/test_multifs_auth.py

```diff
@@ -76,16 +76,28 @@ def setUp(self):
     def test_rw_with_fsname_and_no_path_in_cap(self):
         PERM = 'rw'
         self.captester.write_test_files(self.mounts)
-        keyring_paths = self._create_client(PERM, fsname=True)
-        self.remount_with_new_client(keyring_paths)
+
+        moncap = 'allow r'
+        osdcap = gen_osd_cap_str(((PERM, self.fs1.name),
```

Comment:
FWIW, I _can_ make sense of what `_create_client` helper does without too much effort being put into what the parameters are and what is expected. I'm not 100% clear how this change simplifies stuff, since it essentially un-rolls the helper. If the helper took 10+ arguments with all sort of default values, had an 100+ LOC implementation and returned a non-trivial value (mix of int, str, etc..), I would still have worked towards splitting the helper into understandable reusable code.

In our verbal conversation, you did mention that this change helps to figure what auth caps will be generated. Cannot the auth caps be dumped in the logs for inspection if that is the intend?

Reply:
Yeah, we can rename the `_create_client()` helper to simplify the code instead of putting the code everywhere.

Reply:
> Cannot the auth caps be dumped in the logs for inspection if that is the intend?

We talked about this, while making the first modification the dev won't have access to logs. Because we run tests _after_ modifications have been made.

Reply:
> > Cannot the auth caps be dumped in the logs for inspection if that is the intend?
> 
> We talked about this, while making the first modification the dev won't have access to logs. Because we run tests _after_ modifications have been made.

Won't an elaborated code comment on what the function does help? Furthermore, I did mention another thing - if there is a bug in the way auth caps are generate (say), that's not going to be caught by this change anyway.

Reply:
> > Furthermore, I did mention another thing - if there is a bug in the way auth caps are generate (say), that's not going to be caught by this change anyway.
> 
> Such would caught automatically when auth caps command would reject it. Besides that, bugs can only be caught through logs.

I don't agree. Although with logs, debugging is a lot easier.

> All that is outside the scope of this PR. This PR intend only to make it each step in test more apparent (while reading the test) rather bundling and hidning all the steps inside a method.

What's different with the test case vs (say) MDS daemon code? Are you saying that using helpers in test cases isn't a good idea (even if the helper accept just a handful number of args)?

What's not 100% clear to me with this change, is how _not_ using a helper makes reading a test easier. Is it just me? @ceph/cephfs

Reply:
Anyways, I can work with a compromise (renaming `_create_client()` or splitting it into 2 distinct methods) that we can mutually agree upon.

Reply:
> What I am saying is that hiding (under a layer of abstraction) critical steps of the test is not a good idea.

It *is* a good idea. Those abstractions let developers be concerned about implementation. One can always dig deeper and see what those abstractions do (when required).